tests/smoke_test.sh

```diff
@@ -0,0 +1,229 @@
+
+set -eu
+
+
+### CONSTANT VALUES
+DEFAULT_NODE_COUNT=3
+TEST_IDX="test"
+# Default APP name:
+APP="opensearch"
+OSD="opensearch-dashboard"
+URL=
+OSD_PORT=5601
+INTEGRATOR="data-integrator"
+MODEL=
+
+# Error codes
+ERROR_OPTION=1
+#ERROR_BASIC_HTTP=2
+ERROR_CLUSTER_NOT_GREEN=3
+ERROR_CLUSTER_COUNT_WRONG=4
+ERROR_CREATE_INDEX_FAILED=5
+#ERROR_SHARDS_NOT_STARTED=6
+ERROR_DELETING_IDX_FAILED=7
+ERROR_INDEXING_DOC_FAILED=8
+ERROR_COUNT_INDEX_DOC_FAILED=9
+ERROR_DASHBOARDS_NOT_ACCESSIBLE=10
+################################################
+
+
+function usage() {
+cat << EOF
+usage: smoke_test.sh [OPTIONS]
+To be ran / setup once per cluster.
+-m            (Required)  Model name for the deployment
+-o            (Optional)  Name of the opensearch app to be targeted for these tests. Defaults to "opensearch"
+-u            (Optional)  OpenSearch dashboards URL. Defaults to empty value.
+-d            (Optional)  Name of the opensearch-dashboard app to be targeted for these tests if the URL is not set. Defaults to "opensearch-dashboard"
+-i            (Optional)  Name of the data-integrator app to be targeted for these tests. Defaults to "data-integrator"
+--help                    Shows help menu
+EOF
+
+exit $ERROR_OPTION
+}
+
+while [ $# -gt 0 ]; do
+    case $1 in
+        -m) shift
+            MODEL=$1
+            ;;
+        -o) shift
+            APP=$1
+            ;;
+        -d) shift
+            OSD=$1
+            ;;
+        -u) shift
+            URL=$1
+            ;;
+        -i) shift
+            INTEGRATOR=$1
+            ;;
+        *)
+            usage
+            ;;
+    esac
+    shift
+done
+shift $((OPTIND-1))
+
+
+function run_prechecks() {
+    if ! jq --help > /dev/null 2>&1; then
+        echo "Missing jq command, consider installing it with 'sudo snap install jq --classic'"
+        exit 1
+    fi
+    if ! curl --help > /dev/null 2>&1; then
+        echo "Missing curl command, consider installing it with 'sudo apt install curl'"
+        exit 1
+    fi
+    if ! juju --help > /dev/null 2>&1; then
+        echo "Missing juju command"
+        exit 1
+    fi
+    if ! juju models | grep "$MODEL" > /dev/null 2>&1; then
+        echo "Model ${MODEL} not found in Juju"
+        exit 1
+    fi
+}
+
+
+# Doing the first checks before moving on
+run_prechecks
+
+# Now, we wait for all apps to be active
+for app in $(juju status -m "${MODEL}" --format json | jq -r '.applications | keys[]'); do 
+    juju wait-for application "$app"
+done
+
+OPENSEARCH_IP=$(juju exec --unit "${APP}"/leader -m "${MODEL}" -- unit-get public-address)
+if [ -z "$URL" ]; then
+    URL=$(juju exec --unit "${OSD}"/leader -m "${MODEL}" -- unit-get public-address)
+else
+    OSD_PORT=443
+fi
+
+OPENSEARCH_USERNAME=$(juju run "$APP"/leader get-password --format=json 2>/dev/null | jq -r '. | values[].results.username')
+OPENSEARCH_PASSWORD=$(juju run "$APP"/leader get-password --format=json 2>/dev/null | jq -r '. | values[].results.password')
```

Comment:
We can run this action only once: 
```suggestion
OPENSEARCH_CREDENTIALS=$(juju run "${APP}"/leader get-password --format=json 2>/dev/null)
OPENSEARCH_USERNAME=$(echo "${OPENSEARCH_CREDENTIALS}" | jq -r '. | values[].results.username')
OPENSEARCH_PASSWORD=$(echo "${OPENSEARCH_CREDENTIALS}" | jq -r '. | values[].results.password')
```